src/hci.cpp

```diff
@@ -1270,12 +1270,12 @@ void intOpenDebugMenu(OBJECT_TYPE id)
 		editPosMode = IED_NOPOS;
 		break;
 	case OBJ_FEATURE:
-		for (unsigned i = 0; i < std::min<unsigned>(numFeatureStats, MAXFEATURES); ++i)
+		for (unsigned i = 0, end = std::min<unsigned>(asFeatureStats.size(), MAXFEATURES); i < end; ++i)
 		{
-			apsFeatureList[i] = asFeatureStats + i;
+			apsFeatureList[i] = &asFeatureStats.at(i);
```

Comment:
In situations like this, since out-of-bounds access is already checked, perhaps we should use `[]` instead of `at()`?

Reply:
You are right. I'll fix that.

Reply:
Fixed by [`e06b7d2` (#3588)](https://github.com/Warzone2100/warzone2100/pull/3588/commits/e06b7d2cec31e48bf6e19aa42b3d6cd033344a74)